src/loki/reach.cc

```diff
@@ -5,34 +5,42 @@ using namespace valhalla::baldr;
 namespace valhalla {
 namespace loki {
 
-directed_reach SimpleReach(const DirectedEdge* edge,
-                           uint32_t max_reach,
-                           GraphReader& reader,
-                           const sif::EdgeFilter& edge_filter,
-                           const sif::NodeFilter& node_filter,
-                           uint8_t direction) {
-
-  // no reach to start with
+Reach::Reach() : Dijkstras() {
+  // Mock up the Location struct with the important stuff missing
+  auto* path_edge = locations_.Add()->add_path_edges();
+  path_edge->set_distance(0);
+  path_edge->set_begin_node(false);
+  path_edge->set_end_node(false);
+}
+
+directed_reach Reach::operator()(const DirectedEdge* edge,
```

Comment:
we've changed this implementation to be a 2 phase process. first we changed simple reach from being liberal to conservative so that it stops expanding along any path that could be restricted. if it ended up not finding the appropriate amount of reach we fall back and run the exact reach computation. the key is that most edges are trivially reachable so most of the time the simple conservative reach finds appropriate reachability anyway without having to do the exact (and expensive) check.